### PR TITLE
Added control 1 a & b, 2 a, 6 a, 15 a checks

### PR DESCRIPTION
**Pull request by RobertB-CVP**
The scope of this pull request is to address the following Checks:

[ECMPS Monitoring Plan Check Specifications](https://ecmps.camdsupport.com/documents/ECMPS%20Monitoring%20Plan%20Check%20Specifications%202022%20Q1.pdf)

 CONTROL-1 - Control Parameter Code Valid (Result A, B)
 CONTROL-2 - Control Code Valid (Result A)
 CONTROL-6 - Control Retire Date Valid (Result A)
 CONTROL-15 - Duplicate Unit Control Records (Result A)

### CODE REVIEW

**Review by subrataroyCVP — CHANGES_REQUESTED** [review]
I am creating a PR for MP Checks. Please follow that instead of import checks.

File: src/dtos/unit-control.dto.ts [hunk 1]
```diff
@@ -16,13 +21,24 @@ export class UnitControlBaseDTO {
...
-  @IsNotEmpty()
+  @IsNotEmpty({
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatResultMessage('CONTROL-1-A', {
+        fieldname: args.property,
+        key: KEY,
+      });
+    }
+  })
   @IsInDbValues(
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
  @IsValidCode(
```

File: src/dtos/unit-control.dto.ts [hunk 2]
```diff
@@ -32,12 +48,19 @@ export class UnitControlBaseDTO {
...
-  @IsNotEmpty()
+  @IsNotEmpty({
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatResultMessage('CONTROL-2-A', {
+        fieldname: args.property,
+        key: KEY,
+      });
+    }
+  })
   @IsInDbValues(
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
  @IsValidCode(
```

File: src/dtos/unit-control.dto.ts [hunk 3]
```diff
@@ -32,12 +48,19 @@ export class UnitControlBaseDTO {
...
     `SELECT distinct control_code as "value" FROM camdecmpsmd.vw_unitcontrol_master_data_relationships`,
     {
       message: (args: ValidationArguments) => {
-        return `${args.property} [UNITCONTROL-FATAL-B] The value for ${args.value} in the Unit Control record ${args.property} is invalid`;
+        return `${args.property} The value for ${args.value} in the Unit Control record ${args.property} is invalid`;
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
        return `The value for ${args.value} in the Unit Control record ${args.property} is invalid`;
```

File: src/dtos/unit-control.dto.ts [hunk 4]
```diff
@@ -51,7 +74,7 @@ export class UnitControlBaseDTO {
   @IsOptional()
   @IsIn(['0', '1'], {
     message: (args: ValidationArguments) => {
-      return `${args.property} [UNITCONTROL-FATAL-A] The value for ${args.value} in the Unit Control record  ${args.property} must be string value of "0" or "1"`;
+      return `${args.property} The value for ${args.value} in the Unit Control record  ${args.property} must be string value of "0" or "1"`;
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
      return `The value for ${args.value} in the Unit Control record  ${args.property} must be string value of "0" or "1"`;
```

File: src/dtos/unit-control.dto.ts [hunk 5]
```diff
@@ -64,7 +87,7 @@ export class UnitControlBaseDTO {
   @IsOptional()
   @IsIsoFormat({
     message: (args: ValidationArguments) => {
-      return `${args.property} [UNITCONTROL-FATAL-A] The value for ${args.value} in the Unit Control record ${args.property} must be a valid ISO date format yyyy-mm-dd`;
+      return `${args.property} The value for ${args.value} in the Unit Control record ${args.property} must be a valid ISO date format yyyy-mm-dd`;
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
      return `The value for ${args.value} in the Unit Control record ${args.property} must be a valid ISO date format yyyy-mm-dd`;
```

File: src/dtos/unit-control.dto.ts [hunk 6]
```diff
@@ -77,7 +100,7 @@ export class UnitControlBaseDTO {
...
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
      return `The value for ${args.value} in the Unit Control record ${args.property} must be a valid ISO date format ${DATE_FORMAT}`;
```

File: src/dtos/unit-control.dto.ts [hunk 7]
```diff
@@ -93,7 +116,7 @@ export class UnitControlBaseDTO {
   @IsOptional()
   @IsIn(['0', '1'], {
     message: (args: ValidationArguments) => {
-      return `${args.property} [UNITCONTROL-FATAL-A] The value for ${args.value} in the Unit Control record  ${args.property} must be string value of "0" or "1"`;
+      return `${args.property} The value for ${args.value} in the Unit Control record  ${args.property} must be string value of "0" or "1"`;
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
      return `The value for ${args.value} in the Unit Control record  ${args.property} must be string value of "0" or "1"`;
```

File: src/dtos/unit-control.dto.ts [hunk 8]
```diff
@@ -103,10 +126,18 @@ export class UnitControlBaseDTO {
...
+        fieldname: args.property,
+        date: args.value,
+        key: KEY,
+      });
+    }
+  })
   @IsIsoFormat({
     message: (args: ValidationArguments) => {
-      return `${args.property} [UNITCONTROL-FATAL-A] The value for ${args.value} in the Unit Control record ${args.property} must be a valid ISO date format yyyy-mm-dd`;
+      return `${args.property} The value for ${args.value} in the Unit Control record ${args.property} must be a valid ISO date format yyyy-mm-dd`;
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
      return `The value for ${args.value} in the Unit Control record ${args.property} must be a valid ISO date format ${DATE_FORMAT}`;
```

File: src/monitor-plan-workspace/monitor-plan-checks.service.ts [hunk 9]
```diff
@@ -48,6 +50,23 @@ export class MonitorPlanChecksService {
...
           );
         });
+
+        monitorLocation.unitControls?.forEach(unitControl => {
+          promises.push(
+            new Promise(async (resolve, _reject) => {
+              const results = this.unitControlsChecksService.runChecks(
+                unitControl,
+                monitorLocation.unitRecordId,
+                monitorLocation.id,
```
Comment (subrataroyCVP) [inline src/monitor-plan-workspace/monitor-plan-checks.service.ts]:
```suggestion
                monitorLocation.unitRecordId,
                monitorLocation.id,
```

These wont be provided by user.

**Review by subrataroyCVP — CHANGES_REQUESTED** [review]

File: src/entities/unit-control-codes.entity.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import { BaseEntity, Column, Entity, PrimaryColumn } from 'typeorm';
+
+@Entity({ name: 'camdecmpsmd.vw_unitcontrol_master_data_relationships' })
+export class unitControlCodes extends BaseEntity {
```
Comment (subrataroyCVP) [inline src/entities/unit-control-codes.entity.ts]:
```suggestion
export class UnitControlCode extends BaseEntity {
```

File: src/dtos/unit-control.dto.ts [hunk 11]
```diff
@@ -16,31 +23,49 @@ export class UnitControlBaseDTO {
...
-  )
+  @IsNotEmpty({
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatResultMessage('CONTROL-2-A', {
+        fieldname: args.property,
+        key: KEY,
+      });
+    }
+  })
+  @IsValidCode(unitControlCodes, {
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
  @IsValidCode(UnitControlCode, {
```

**Review by subrataroyCVP — CHANGES_REQUESTED** [review]
Need Small Change.

File: src/dtos/unit-control.dto.ts [hunk 12]
```diff
@@ -16,31 +23,49 @@ export class UnitControlBaseDTO {
...
-  )
+  @IsNotEmpty({
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatResultMessage('CONTROL-1-A', {
+        fieldname: args.property,
+        key: KEY,
+      });
+    }
+  })
+  @IsValidCode(ParameterCode, {
```
Comment (subrataroyCVP) [inline src/dtos/unit-control.dto.ts]:
```suggestion
  @IsValidCode(ParameterCode, {
```

Previously it was doing look up for parameter on camdecmpsmd.vw_unitcontrol_master_data_relationships.
Now you are doing against parameterCode table. Also in the checks it says `Locate ParameterCode in the Control Code lookup table.`

**Review by subrataroyCVP — CHANGES_REQUESTED** [review]

File: src/dtos/unit-control.dto.ts (hunk 12 above)
Comment (subrataroyCVP) [reply src/dtos/unit-control.dto.ts]: Follow the DB Script.

**Review by jordanfontainecvp — DISMISSED** [review]

**Review by subrataroyCVP — APPROVED** [review]

**Pull request merged**